modules/std.lobster

```diff
@@ -119,21 +119,30 @@ def qsort(xs, lt):
 def qsort_in_place(xs, lt):
     def rec(s, e):
         let l = e - s
-        if l > 1:
-            let pivot = xs[s]
-            var sp = s + 1
-            var ep = e
-            while sp < ep:
-                let c = xs[sp]
-                if lt(c, pivot):
-                    xs[sp - 1] = xs[sp]
-                    sp++
-                else:
-                    xs[sp] = xs[--ep]
-                    xs[ep] = c
-            xs[--sp] = pivot
-            rec(s, sp)
-            rec(ep, e)
+
+        switch l:
+            case 0, 1:
+                return
+
+        if l <= 32:
+            insertion_sort(slice(xs, s, l), lt)
```

Comment:
I'm sure this might be quicker, but it says `qsort` on the box, so it should do qsort :) Lets not mix sorting algos. Besides being a stdlib, its also an example, so certain level of elegance is good. And small is good too, seeing as how these functions typically all get specialized.

Reply:
Agree, again this is just me experimenting, and shared only to demonstrate a crash that I'm having trouble sorting out (pun very much intended)

Reply:
Although, it's very rare in practice for anything calling itself "quicksort" to be "quicksort all the way down"; that's usually only in teaching example code ;-)